### ISSUE TITLE
Any item outside of the tsconfig.include option from the 'src' root will move dts files around in dist

### ISSUE THREAD

**Issue by VividLemon** [qmhc/vite-plugin-dts#191]
### Describe the bug

I am currently in the process of refactoring an app to make it more consistent with the output of `npm create vue@3`... Something important that occurs when using this starter template is that env.d.ts, which contains the vite reference types, it is in the project root, rather than the src.

Why is this an issue? Seemingly any item that is in the tsconfig.include option that exists outside the typical 'src', will cause the dist output to act strangely. 

Instead of the dts files being loaded onto the ./dist root, ex ./dist/components/foo.d.ts will instead become ./dist/src/components/foo.d.ts

One can see how this would become a problem. 

Now, I'm not certain how ts decides to use types, especially in this context. I'm not sure if you can set your package.json "types" option to simply become the new path (eg: `./dist/src/entry.d.ts`) and if that will provide the same level of type inference to a user of the library.

If it is a simple change of the 'types' option in package.json, it's a fairly simple fix to change the path. However, one could also explicitly exclude it, or move env.d.ts into the src. Please, let me know if the changing package.json types option is sufficient, I'm curious as I'm not sure how exactly ts infers everything and would like to know.

If you feel comfortable, here is a zip, just mess with the options on the vite.config.ts
[vue-project.zip](https://github.com/qmhc/vite-plugin-dts/files/11011526/vue-project.zip)

I'm not exactly sure if this is really a 'bug' issue... It seems fully intentional, for good reason. However, it seems strange. Perhaps a note or some guidance could clear up the confusion. Or perhaps it's just me, perhaps the package.json 'types' option changing from the previous `types: './dist/entry.d.ts'` to `types: './dist/src/entry.d.ts'` is something that would be intuitive to another user, idk.

### Reproduction

I apologize, I omitted it, it's fairly straightforward, review 'steps to reproduce'

### Steps to reproduce

npm create vue@3

install vite-plugin-dts and configure

build and review dist

(It's fairly simple, so reproduction link is omitted)

If you then proceed to `dts({ exclude: ['env.d.ts'] })`, it magically starts working (In the npm create vue@3 route, you will likely want to do `tsConfigFilePath: 'tsconfig.app.json'`) for:
```ts
    dts({
      exclude: ["env.d.ts"],
      tsConfigFilePath: "tsconfig.app.json",
    }),
```

### System Info

```shell
System:
    OS: Linux 5.15 Linux Mint 21.1 (Vera)
    CPU: (12) x64 AMD Ryzen 5 3600X 6-Core Processor
    Memory: 8.20 GB / 15.55 GB
    Container: Yes
    Shell: 5.8.1 - /usr/bin/zsh
  Binaries:
    Node: 19.4.0 - /run/user/1000/fnm_multishells/71773_1679239925333/bin/node
    Yarn: 1.22.19 - /run/user/1000/fnm_multishells/71773_1679239925333/bin/yarn
    npm: 9.2.0 - /run/user/1000/fnm_multishells/71773_1679239925333/bin/npm
  Browsers:
    Firefox: 111.0
  npmPackages:
    @vitejs/plugin-vue: ^4.0.0 => 4.0.0 
    vite: ^4.0.0 => 4.1.4 
    vite-plugin-dts: ^1.7.1 => 1.7.3
```

### Validations

- [X] Read the [FAQ](https://github.com/qmhc/vite-plugin-dts#faq).
- [X] Check that there isn't [already an issue](https://github.com/qmhc/vite-plugin-dts/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by qmhc** [comment]
Do you try to use the plugin@v2 if that is what you expect?

And I think this is something about `copyDtsFiles` option.

**Comment by VividLemon** [comment]
This is when using @2. I'm not certain if the same issue exists on @1 

I'm not sure if copyDtsFiles is apart of the problem. For a lot of my testing, the option made no affect to the dist

**Comment by qmhc** [comment]
Fixed in `3.0.0`.

In some cases, if `entryRoot` is specified, some declaration files maybe outside `outDir`. Now default strictly restrict declaration files inside the `outDir`, and add `strictOutput` option to control this behavior.
